### PR TITLE
[BE][DTensor] merge random init test to test_random_ops.py

### PR DESCRIPTION
**Pull request by XilunWu**
Random Ops behavior becomes different on CUDA device and others in PR https://github.com/pytorch/pytorch/pull/98199 . As a result, test of DTensor random initialization in test_init.py was skipped. This PR fixes it by having different testing assertion for different types of device and also merges existing duplicated test.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99014
* __->__ #98874

### CODE REVIEW

File: test/distributed/_tensor/test_random_ops.py [hunk 1]
```diff
@@ -26,6 +27,63 @@
...
+            dtensor = dtensor.redistribute(device_mesh, [Replicate()])
+            local_tensor_gathered = dtensor.to_local()
+
+            # compare with local tensors from other ranks
+            for other_rank in range(self.world_size):
+                if self.rank != other_rank:
+                    slice_idx = [
+                        slice(input_size[0]),
+                        slice(
+                            other_rank * input_size[1], (other_rank + 1) * input_size[1]
```
Comment (wanchaol) [inline test/distributed/_tensor/test_random_ops.py:69]: what if `other_rank` is the last rank, would it still generate the right shape?

**Review by wanchaol — APPROVED** [review]
looks good! Only have one question

File: test/distributed/_tensor/test_random_ops.py (hunk 1 above)
Comment (XilunWu) [reply test/distributed/_tensor/test_random_ops.py:69]: Yes. I tried the following example:
```
>>> import torch
>>> tensor = torch.arange(32).reshape(4, 8)
>>> last_rank = 3
>>> slice_idx = [slice(4), slice(last_rank * 2, (last_rank + 1) * 2)]
>>> tensor[slice_idx]
tensor([[ 6,  7],
        [14, 15],
        [22, 23],
        [30, 31]])
```

**Comment by malfet** [comment]
https://github.com/pytorch/pytorch/commit/49cd650e2bd6d84ee936adac62c379bc8b67dd53 has been merged to trunk, so closing

**Pull request closed without merging**